build.py

```diff
@@ -114,25 +142,28 @@ def build(config_name, configure_parameters, build_parameters):
     print(f"Built configuration {config_name} successfully.")
 
 
-def main():
-    config_names = []
-    build_parameters = []
-    for arg in sys.argv[1:]:
-        if arg == "--help" or arg == "-h":
-            print_usage()
-            sys.exit(0)
-        elif arg == "--debug":
-            config_names.append(DEBUG_CONFIG_NAME)
-        elif arg == "--all":
-            config_names.extend(sorted(CONFIGS.keys()))
-        elif arg in CONFIGS:
-            config_names.append(arg)
+def split_args(args):
+    args.config_names = []
+    args.build_parameters = []
+    for arg in args.arguments:
+        if arg in CONFIGS:
+            args.config_names.append(arg)
         else:
-            build_parameters.append(arg)
-    if not config_names:
-        config_names.append(DEFAULT_CONFIG_NAME)
-    for config_name in config_names:
-        build(config_name, CONFIGS[config_name], build_parameters)
+            args.build_parameters.append(arg)
+
+    if args.debug:
+        args.config_names.append(DEBUG_CONFIG_NAME)
+    if args.all:
+        args.config_names.extend(sorted(CONFIGS.keys()))
+    if not args.config_names:
+        args.config_names.append(DEFAULT_CONFIG_NAME)
+
```

Comment:
It might make sense to eliminate duplicates in `args.config_names`. If I call `./build.py debug debug`, it will build the debug config two times. (Of course, the second time it will be very fast since everything is built already.)

Reply:
Trying to build it twice is exactly what the user asks for with this call. I think it's not a good feature to try to "optimize" this.